libs/ledger-live-common/src/exchange/swap/hooks/useSwapTransaction.ts

```diff
@@ -25,18 +28,49 @@ export const selectorStateDefaultValues = {
 
 export type SetExchangeRateCallback = (exchangeRate?: ExchangeRate) => void;
 
-export const useFromAmountError = (
-  errors: Record<string, Error | undefined>,
+export const useFromAmountStatusMessage = (
+  { account, parentAccount, status, transaction }: Result<Transaction>,
+  // The order of errors/warnings here will determine the precedence
+  statusToInclude: string[],
 ): Error | undefined => {
-  const fromAmountError = useMemo(() => {
-    const [error] = [errors?.gasPrice, errors?.amount]
+  const statusEntries = useMemo(
+    () => statusToInclude.map(s => (status.errors || status.warnings)?.[s]),
```

Comment:
Can we find something more descriptive than 's'?

Reply:
Of course! 👍